### ISSUE TITLE
[BUG] Can't upgrade from master version to master version

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#3444]
**Describe the bug**

Upgrading to the same version fails (with the current master code).

**To Reproduce**
1. Build an ISO with the current master code (6fd9a5bb3b99a329101ec70f657a30e24906035f) and deploy a cluster.
2. Upgrade with the ISO.
3. The upgrade fails with

    ![image](https://user-images.githubusercontent.com/1691518/218121471-fb0d21b0-c7e2-41cf-ba46-e49d08978634.png)

**Expected behavior**

The upgrade should proceed and succeed.

**Support bundle**

**Environment**
 - Harvester ISO version: 6fd9a5bb3b99a329101ec70f657a30e24906035f
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by starbops** [comment]
By taking several rounds of testing, we noticed that the upgrade could only pass the first and second phases, i.e., `ImageReady` and `RepoReady`.

```
  status:
    conditions:
    - lastUpdateTime: "2023-02-13T03:45:41Z"
      reason: 'Get "http://upgrade-repo-hvst-upgrade-rxcvc.harvester-system/harvester-iso/harvester-release.yaml":
        dial tcp 10.53.200.90:80: connect: connection refused'
      status: "False"
      type: Completed
    - lastUpdateTime: "2023-02-13T03:44:44Z"
      status: "True"
      type: ImageReady
    - lastUpdateTime: "2023-02-13T03:45:41Z"
      status: "True"
      type: RepoReady
```

Soon it failed at the beginning of the `NodesPrepared` phase: the upgrade controller tried to get the information from the repo VM it spawned up but failed with the "connection refused" error, thus it marked the upgrade as failed by setting the `Completed` condition to "False".

https://github.com/harvester/harvester/blob/6fd9a5bb3b99a329101ec70f657a30e24906035f/pkg/controller/master/upgrade/upgrade_controller.go#L168-L172

https://github.com/harvester/harvester/blob/6fd9a5bb3b99a329101ec70f657a30e24906035f/pkg/controller/master/upgrade/upgrade_repo.go#L418-L429

Then the repo VM is purged by the upgrade controller immediately.

https://github.com/harvester/harvester/blob/6fd9a5bb3b99a329101ec70f657a30e24906035f/pkg/controller/master/upgrade/upgrade_controller.go#L143-L146

To further justify the healthiness and readiness of the repo VM, we scaled down the harvester Deployment to 0 as soon as the repo VM was created. By execing into the harvester-webhook Pod (to grab a shell for example), we can curl the URL (http://<repo-svc>.harvester-system/harvester-iso/harvester-release.yaml) successfully. So the repo VM is good.

**Comment by starbops** [comment]
It is the OS bump that causes the problem. I built two images, one with harvester/harvester-installer [6979eac](https://github.com/harvester/harvester-installer/commit/6979eac93ceb478d6169a72bf61c284fa425dfa5) (the commit right before the os bump) and the other with [d8930db](https://github.com/harvester/harvester-installer/commit/d8930db4be133cf516ee931629dde6a7679b09b2) (the OS bump commit), only the latter failed the upgrade. Both images are built with harvester/harvester master code.

**Comment by starbops** [comment]
Adding a simple retry can work around this.

```
		repoInfo, err := repo.getInfo()
		if err != nil {
			h.upgradeController.EnqueueAfter(upgrade.Namespace, upgrade.Name, 10*time.Second)
			//setUpgradeCompletedCondition(toUpdate, StateFailed, corev1.ConditionFalse, err.Error(), "")
			return h.upgradeClient.Update(toUpdate)
		}
```

Some of the internal components' behavior is slightly different in the new OS. The root cause is still not clear.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-89b3108b-head`. Close this issue.

### Result
We can upgrade prebuild master release to the same master version again correctly 
![image](https://user-images.githubusercontent.com/29251855/219661362-5fbab646-94d4-47e1-ad69-a83e3664f0cf.png)

We can proceed the preloading stage
![image](https://user-images.githubusercontent.com/29251855/219655376-3b4ab633-f57a-4ae4-b963-06b195e0b968.png)

### Test Information
* Test Environment: 3 nodes harvester on local kvm machine
* Harvester version: master-89b3108b-head

### Verify Steps
1. Prepare prebuild v1.1.2 master release
2. Provision a 3 nodes Harvester cluster 
3. Prepare offline upgrade image
4. Upgrade to the same master release 
![image](https://user-images.githubusercontent.com/29251855/219652818-a9e5c3ec-12de-48b9-a92c-bb873062f481.png)
